core/vtk/ttkPlanarGraphLayout/ttkPlanarGraphLayout.cpp

```diff
@@ -144,3 +151,73 @@ int ttkPlanarGraphLayout::RequestData(vtkInformation *request,
 
   return 1;
 }
+
+template <class dataType>
+int ttkPlanarGraphLayout::mergeTreePlanarLayoutCallTemplate(
+  vtkUnstructuredGrid *treeNodes,
+  vtkUnstructuredGrid *treeArcs,
+  vtkUnstructuredGrid *output) {
+  int verbose = 0;
+  MergeTree mergeTree = makeTree<dataType>(treeNodes, treeArcs);
+  FTMTree_MT *tree = &(mergeTree.tree);
+  // tree->printTree2();
+
+  computePersistencePairs<dataType>(tree);
+
+  std::vector<std::vector<int>> treeNodeCorrMesh(1);
+  treeNodeCorrMesh[0] = std::vector<int>(tree->getNumberOfNodes());
+  for(unsigned int j = 0; j < tree->getNumberOfNodes(); ++j)
+    treeNodeCorrMesh[0][j] = j;
+
+  ttkMergeTreeVisualization visuMaker;
+  visuMaker.setPlanarLayout(true);
+  visuMaker.setOutputSegmentation(false);
+  visuMaker.setBranchDecompositionPlanarLayout(BranchDecompositionPlanarLayout);
+  visuMaker.setBranchSpacing(BranchSpacing);
+  visuMaker.setImportantPairs(ImportantPairs);
+  visuMaker.setImportantPairsSpacing(ImportantPairsSpacing);
+  visuMaker.setNonImportantPairsSpacing(NonImportantPairsSpacing);
+  visuMaker.setNonImportantPairsProximity(NonImportantPairsProximity);
+  visuMaker.setVtkOutputNode(output);
+  visuMaker.setVtkOutputArc(output);
+  visuMaker.setTreesNodes(treeNodes);
+  visuMaker.setTreesNodeCorrMesh(treeNodeCorrMesh);
+  visuMaker.makeTreesOutput<dataType>(tree, verbose);
+
+  return 1;
+}
+
+int ttkPlanarGraphLayout::mergeTreePlanarLayoutCall(
+  vtkInformation *request,
+  vtkInformationVector **inputVector,
+  vtkInformationVector *outputVector) {
+
+  vtkUnstructuredGrid *treeNodes
+    = vtkUnstructuredGrid::GetData(inputVector[0], 0);
+  vtkUnstructuredGrid *treeArcs
+    = vtkUnstructuredGrid::GetData(inputVector[0], 1);
+  auto output = vtkUnstructuredGrid::GetData(outputVector);
+  int dataTypeInt
+    = treeNodes->GetPointData()->GetArray("Scalar")->GetDataType();
+
+  /*treeNodes->PrintSelf(std::cout, vtkIndent(2));
+  treeArcs->PrintSelf(std::cout, vtkIndent(2));*/
+
+  int res = 0;
+
+  switch(dataTypeInt) {
+    vtkTemplateMacro(res = mergeTreePlanarLayoutCallTemplate<VTK_TT>(
```

Comment:
be careful, `vtkTemplateMacro` are (or will be soon) deprecated. We will need to replace this code to use dispatcher in the futur

Reply:
Indeed, I was not aware of that! Perhaps I will keep this for the moment until TTK moves fully to the new dispatcher way.

Reply:
ok!

Reply:
@CharlesGueunet 
that'd be great if you could let us know more about the deprecation of `vtkTemplateMacro`. it is used extensively throughout ttk (for each filter actually).

Reply:
I see dark clouds at the horizon. Especially since we do have the same logic to templatize triangulations. I guess this deserves a new issue and a bigger discussion.

Reply:
weren't we looking for an excuse to throw a hackathon?